src/apps/backend/modules/comment/internal/comment-reader.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  Comment,
+  CommentNotFoundError,
+  GetCommentParams,
+  GetCommentsParams,
+} from '../types';
+
+import CommentUtil from './comment-util';
+import CommentRepository from './store/comment-repository';
+
+export default class CommentReader {
+  public static async getCommentsForTask(
+    params: GetCommentsParams
+  ): Promise<Comment[]> {
+    const commentsDb = await CommentRepository.find({
+      task: params.taskId,
+    }).populate('account', 'firstName lastName username');
```

Comment:
This is not OK.

By directly inspecting the `account` database, you are violating the boundary definition aka now the comment module is taking dependency on something very internal to `account` module. Anything that is not exposed by AccountService or its types.ts file is considered internal and its not Ok to take dependency on it.